src/lib.rs

```diff
@@ -10,25 +14,14 @@ pub fn matcher<'a>(pattern: &'a str, url: &'a str) -> Option<HashMap<String, &'a
     let expr = util::expression(pattern);
     let re = util::regexp(&expr);
 
-    if re.is_match(url) {
-
-        let caps = re.captures(url).unwrap();
-
-        for (index, key) in re.capture_names().enumerate() {
-
-            if key.is_some() {
-                let k = String::from(key.unwrap());
-                let v = caps.get(index).unwrap().as_str();
-
-                map.insert(k, v);
-
-            }
+    let caps = re.captures(url)?;
```

Comment:
[`?` operator][1] can be used to unwrap this value because [`Regex::captures`][2] returns `Option<...>`, and `matcher` function which contains this line also returns `Option<...>`.

This is much useful while dealing with `Result<T, E>` to handle errors.

[1]: https://doc.rust-lang.org/reference/expressions/operator-expr.html#the-question-mark-operator
[2]: https://docs.rs/regex/0.2.10/regex/struct.Regex.html#method.captures